### DISCUSSION TITLE
Codium Remote

### DISCUSSION THREAD

**Discussion by notpushkin** [VSCodium/vscodium#832] [Ideas]
As discussed in #196 and #240, Microsoft's proprietary Remote extensions aren't usable in OSS builds (insiders versions might be, though – see https://github.com/microsoft/vscode-remote-release/issues/574). Perhaps we could look into making an open source replacement of our own, based on the stubs that _are_ provided in the source code?

**Comment by matthewpwatkins** [comment]
I would love to see this as my current company does not approve the pre-built binaries from MS so we all use VSCodium and would love to use Remote extension functionality.

**Comment by stripedpajamas** [comment]
Yeah that would be super cool. I don't have the capacity to contribute to that currently, but would help out where possible.

If insiders currently supports unlicensed builds, it may mean that the standard version will support them in the near future; also a hope is to get VSCodium automated builds of the insider's version in the near future (#148). That would potentially unblock using the remote extensions without a big effort building from piecemeal.

**Comment by alexandrudsc** [comment]
hello, any news on this? I love vscodium as is, but I would like to use this as well

**Comment by kmantel** [comment]
In the meantime, [ponyssh](https://marketplace.visualstudio.com/items?itemName=thingalon.pony-ssh) has done what I needed

**Comment by kiprasmel** [comment]
@kmantel 
> In the meantime, [ponyssh](https://marketplace.visualstudio.com/items?itemName=thingalon.pony-ssh) has done what I needed

Yeah that might be cool, but we'd still prefer the actual remote development extensions.

Any news here?

**Comment by notpushkin** [comment]
@sarpik Pony SSH could be a good start though? :thinking:

**Comment by dalisoft** [comment]
Pony SSH asks my remote server SSH password even i loaded it to agent, so, looks like insecure( I really want use SSH remote development

**Comment by laurianops** [comment]
any news of remote access to the remote-ssh ?

**Comment by gyzerok** [comment]
I am interested in contributing my time to create an ssh extension. If anybody has any pointers as to where I should start, they are very welcome.

My initial understanding is that I will need to figure out how to start vscode headless to use it for the "server" and develop an extension which will register different providers remotely.

**For future readers**
Please, avoid `any news?` comments. You are just pinging everybody in this thread for no reason. It is likely that people will get annoyed and unsubscribe, so even if there will be news they won't update this issue.

**Comment by Lcchy** [comment]
I was looking for a solution to this and found something on the way but I'm not certain about it.
Can anyone tell me if it's a good idea to use [sshfs](https://github.com/libfuse/sshfs) in order to mount the remote server into the local working directory and then use VSCodium to browse/edit the files ? 
It seems like it's too good to be true, why would one need any extension then ?

**Comment by notpushkin** [comment]
@Lcchy Mainly for the debugger I think. Other things (version control and code jump to dependencies) most of the time work fine with just files, but debugging requires running a special process on the server (for some languages it's a separate script, e. g. [debugpy](https://github.com/microsoft/debugpy) for Python, for others it's just a runtime option, e. g. `node --inspect`) and port forwarding (so that nobody else can “debug” your code :) to work.

**Comment by notpushkin** [comment]
Another good starting point might be to try to hook up a vscode-ish server ([GitPod](https://github.com/gitpod-io/gitpod), [code-server](https://github.com/cdr/code-server)) to the Codium UI.

Also related: https://github.com/cdr/code-server/issues/1315

**Comment by bpmct** [comment]
> Another good starting point might be to try to hook up a vscode-ish server ([GitPod](https://github.com/gitpod-io/gitpod), [code-server](https://github.com/cdr/code-server)) to the Codium UI.

Would this mean that Codium would install code-server on a remote resource and then proxy that connection through the Codium window? Interesting!

**Comment by eabase** [comment]
so can e remote at this point? ❓

**Reply by notpushkin** [reply]
Not really. I guess nobody's motivated enough to deal with it :-(

**Comment by vkottler** [comment]
Looking at the [Proprietary Extensions](https://github.com/VSCodium/vscodium/blob/master/DOCS.md#proprietary-extensions) documentation section:

> In some cases, the above change won't help because the extension is hard-coded to only work with the official Visual Studio Code product.

The need for a one-off solution does not seem like the problem here. A path forward is needed _in general_ for extensions in this category, since they're some of the most important extensions for doing any kind of serious development work.

This seems like a solve-able problem that maybe warrants discussion over in the official repository. At the very least I think a Microsoft developer could tell you how to go about this and whether or not they'd consider making changes on their end (they should... this is very silly).

I check in every now and then to see if Code and VSCodium are usable yet (compared to multi-decade-old editors like Vim), but it doesn't seem this is getting any attention. At the same time, VSCodium is not advertising itself as unmaintained or deprecated. Maybe it should.

**Reply by daiyam** [reply]
> This seems like a solve-able problem that maybe warrants discussion over in the official repository. At the very least I think a Microsoft developer could tell you how to go about this and whether or not they'd consider making changes on their end (they should... this is very silly).

It is only a license issue. 

> At the same time, VSCodium is not advertising itself as unmaintained or deprecated. Maybe it should.

You know how to piss off people. Congrats!

The only viable solution for those extensions is to develop open-source alternatives.

**Comment by ANaaame** [comment]
ssh-fs works for me.

After adding this to setting.json, ssh-fs works well !
`    "sshfs.configs": [
        {
            "name": "a name",
            "host": "IP Address",
            "label": "a label",
            "root": "folder path that you want to open",
            "username": "username",
            "password":"password"
        }
    ],`

**Comment by kisg** [comment]
Someone started the implementation: https://github.com/xaberus/vscode-remote-oss

**Comment by Cyberbeni** [comment]
I am using this `product.json` for connecting to the MS store and downloading their proprietary extensions:
```json
{
  "nameLong": "Visual Studio Code",
  "extensionsGallery": {
    "serviceUrl": "https://marketplace.visualstudio.com/_apis/public/gallery",
    "cacheUrl": "https://vscode.blob.core.windows.net/gallery/index",
    "itemUrl": "https://marketplace.visualstudio.com/items"
  }
}
```

**Comment by raj-magesh** [comment]
https://github.com/jeanp413/open-remote-ssh

This extension satisfies my requirements, in case others in this thread haven't seen it yet.
